### PR TITLE
Implement null-safety

### LINKED ISSUE #122: Support null safety
This is a request to add support for [null safety](https://dart.dev/null-safety) to `package:intercom_flutter`.
We depend on your awesome package, so would be great to have null safety enabled.

The Dart/Flutter team already encourages publishing the migrated packages: See [this blog post](https://medium.com/dartlang/preparing-the-dart-and-flutter-ecosystem-for-null-safety-e550ce72c010).

See the [migration guide](https://dart.dev/null-safety/migration-guide) for details about enabling null safety.

### PR DESCRIPTION
**Pull request by IchordeDionysos**
Implements new Dart's null-safety feature for `intercom_flutter`.

Fixes #122.

### CODE REVIEW

File: example/pubspec.yaml [hunk 1]
```diff
@@ -14,4 +15,4 @@ flutter:
   uses-material-design: true
 
 environment:
-  sdk: '>=2.10.0 <3.0.0'
\ No newline at end of file
+  sdk: '>=2.12.0-133.7.beta <3.0.0'
```
Comment (Zazo032) [inline example/pubspec.yaml]:
```suggestion
  sdk: '>=2.12.0 <3.0.0'
```

File: .gitignore [hunk 2]
```diff
@@ -8,3 +8,7 @@ pubspec.lock
 build/
 .idea/
 android/bin/
+
+example/.flutter-plugins-dependencies
+
+example/ios/Flutter/flutter_export_environment.sh
```
Comment (Zazo032) [inline .gitignore]: Remove blank lines

File: pubspec.yaml [hunk 3]
```diff
@@ -22,5 +23,5 @@ flutter:
         pluginClass: IntercomFlutterPlugin
 
 environment:
-  sdk: ">=2.7.0 <3.0.0"
+  sdk: '>=2.12.0-133.7.beta <3.0.0'
```
Comment (Zazo032) [inline pubspec.yaml]:
```suggestion
  sdk: '>=2.12.0 <3.0.0'
```

**Review by Zazo032 — CHANGES_REQUESTED** [review]
Thanks for the contribution @IchordeDionysos!

I left some minor nits, and this should be ready to be merged when those are fixed and #126 lands.

File: pubspec.yaml [hunk 4]
```diff
@@ -11,6 +11,7 @@ dependencies:
 dev_dependencies:
   flutter_test:
     sdk: flutter
+  collection: ^1.15.0-nullsafety.4
```
Comment (Zazo032) [inline pubspec.yaml]:
```suggestion
  collection: ^1.15.0
```

**Comment by IchordeDionysos** [comment]
@Zazo032 I've adjusted the PR and also updated the Flutter version that is used for the automated tests 👍

**Comment by IchordeDionysos** [comment]
Ahh okay sorry :D Did not look into your PR yet 🌝  Can use your changes :)
But now we at least know that my changes work :D

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -22,6 +22,6 @@ jobs:
         java-version: '12.x'
     - uses: subosito/flutter-action@v1
       with:
-        flutter-version: '1.22.3'
+        flutter-version: '2.0.1'
```
Comment (Zazo032) [inline .github/workflows/main.yml]:
```suggestion
        channel: 'stable'
```

**Review by Zazo032 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**